packages/css/src/types.ts

```diff
@@ -465,24 +464,37 @@ export interface VariantProperty {
    *
    * @see https://styled-system.com/variants
    */
-  variant: string
+  variant?: string
 }
 
-export interface UseThemeFunction {
-  (theme: any): Exclude<ThemeUIStyleObject, UseThemeFunction>
+export interface ThemeDerivedStyles {
+  (theme: Theme): Exclude<ThemeUIStyleObject, ThemeDerivedStyles>
 }
 
+export type Label = {
+  label?: string
+}
+
+export interface CSSOthersObject {
+  // we want to match CSS selectors
+  // but index signature needs to be a supertype
+  // so as a side-effect we allow unknown CSS properties (Emotion does too)
+  [k: string]: StylePropertyValue<string | number> | undefined
+}
+
+export interface ThemeUICSSObject
+  extends ThemeUICSSProperties,
+    CSSPseudoSelectorProps,
+    CSSOthersObject,
+    VariantProperty,
+    Label {}
+
 /**
  * The `ThemeUIStyleObject` extends [style props](https://emotion.sh/docs/object-styles)
  * such that properties that are part of the `Theme` will be transformed to
  * their corresponding values. Other valid CSS properties are also allowed.
  */
-export type ThemeUIStyleObject =
-  | ThemeUICSSProperties
-  | CSSPseudoSelectorProps
-  | CSSSelectorObject
-  | VariantProperty
-  | UseThemeFunction
+export type ThemeUIStyleObject = ThemeUICSSObject | ThemeDerivedStyles
```

Comment:
You can see I removed `CSSSelectorObject` and merged all objects into `ThemeUICSSObject`.

The naming is not perfect here.

Reply:
I hate this (being too permissive) but you're absolutely right that it should aim to be least confusing in the event of an error. I *do* feel it should enforce correctness but unless we can have both... errors that make sense are probably more valuable than forbidding properties that will just be ignored anyway. 🤷‍♂️

Reply:
I noticed later today, that this isn't too permissive.
There are variables in CSS 🤦
Arbitrary property names (we can't typecheck the leading dash) with values are 100% legit and useful CSS.

Reply:
Oh, good point. Then I guess we're set! 😄